File: Project4/city.py
import math

class City:

	id = 5
	x = 1.2
	y = 1.2
	
	# using id = -2 for children cities unless necessary
	# defaulting x and y coordinates to a random location to account for children cities, otherwise they're given
	def __init__(self, idin = -2, xin = -3, yin = -4):
		self.id = idin
		self.x = xin
		self.y = yin
		self.visited = 0
		
	def __str__(self):
		return str(self.id)
	
	def getX(self):
		return self.x
	
	def getY(self):
		return self.y
		
	def getID(self):
		return self.id
		
#	def __getitem(self):
#		return {x: self.x, y, self.y}
		
	def distanceTo(self, city):
		#print "distanceTo: type of city is: "
		#print city.__class__
		
		x2 = city.x
		y2 = city.y
	
		x2 -= self.x
		y2 -= self.y
		
		x2 *= x2
		y2 *= y2
		
		return math.sqrt(x2 + y2)

# boston = City(1,2.5,3.5)

File: Project2/city.cpp
#include "city.h"
#include "math.h"
#include <iostream>
#include <fstream>

using namespace std;

city::city (int idin, float xin, float yin) {
	id = idin;
	x = xin;
	y = yin;
}

/* city::city (int idin, float xin, float yin, float distance, city parent) {
	id = idin;
	x = xin;
	y = yin;
	distance = 0.0;
	parent = NULL;
} */

city getParent (int id) {
	
}

// x2 is x-distance from 2nd to 1st city
// x comes from [i]th element, x2 is [i-1]
float city::getDistance(city cityPtr) {
	 
	float x2 = cityPtr.getX();
    float y2 = cityPtr.getY();
	x2 -= x;
    y2 -= y;
    
    x2 *= x2;
    y2 *= y2;
  /*  cout<<"x is "<<x<<"\n--------\n";
    cout<<"y is "<<y<<"\n--------\n";    
	cout<<"x2 is "<<x2<<"\n--------\n";
	cout<<"y2 is "<<y2<<"\n--------\n";*/
    return sqrt(x2 + y2);
}

void city::setX(float xin) {
	x = xin;
}

void city::setY(float yin) {
	y = yin;
}

float city::getX()
{
    return x;
}

/*
x2 from 2nd, which is [i]
x from 1st, which is [i-1]


*/
float city::getY() 
{
    return y;
}

int city::getID()
{
    return id;
}
